File: CS50/Core/pset6/cash/cash.py
from cs50 import get_float

# Get amount of change owed from user, value greater than 0
change = -1
while change < 0:
    change = get_float("Change owed: ")

q = 0
d = 0
n = 0
p = 0
    
# Calculate quarters
while change > 0.24:
    q += 1
    change -= 0.25
    change = round(change, 2)
    
# Calculate dimes
while change > 0.09:
    d += 1
    change -= 0.10
    
# Calculate nickles
while change > 0.04:
    n += 1
    change -= 0.05    

# Calculate pennies
while change > 0:
    p += 1
    change -= 0.01  
    
print(q + d + n + p)    

File: Courses/CS50/Core/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    printf("Welcome to the super awesome change calculator 1.0!\n");
    float change;
    int q = 0;
    int d = 0;
    int n = 0;
    int p = 0;
    int coins = 0;

    //Get float for change owed, postive numbers only
    do
    {
        change = get_float("How much change is due?\n");
    }
    while (change < 0);

    //Change into cents
    int cents = round(change * 100);

    //number of quarters
    while (cents > 24)
    {
        q = q + 1;
        cents = cents - 25;
    }
    //printf("Quarters: %i\n", q);

    //number of dimes
    while (cents > 9)
    {
        d = d + 1;
        cents = cents - 10;
    }
    //printf("Dimes: %i\n", d);

    //number of nickles
    while (cents > 4)
    {
        n = n + 1;
        cents = cents - 5;
    }
    //printf("Nickles: %i\n", n);

    //number of pennies
    while (cents > 0)
    {
        p = p + 1;
        cents = cents - 1;
    }
    //printf("Pennies: %i\n", p);

    //Total number of coins
    coins = q + d + n + p;
    printf("Minimum coins: %i\n", coins);

}